pages/apim/3.x/v4-beta/v4-beta-event-native-apim-user-guide.adoc

```diff
@@ -0,0 +1,243 @@
+
+[source json]
+----
+{
+    "name": "Data Ingestion to Kafka",
+    "apiVersion": "1.0",
+    "definitionVersion": "4.0.0",
+    "type": "async",
+    "description": "Data Ingestion to Kafka",
+    "listeners": [
+        {
+            "type": "http",
+            "paths": [
+                {
+                    "path": "/data/ingestion/kafka"
+                }
+            ],
+            "entrypoints": [
+                {
+                    "type": "http-post",
+                    "configuration": {
+                        "requestHeadersToMessage": false
+                    }
+                }
+            ]
+        }
+    ],
+    "endpointGroups": [
+        {
+            "name": "default",
+            "type": "kafka",
+            "endpoints": [
+                {
+                    "name": "default",
+                    "type": "kafka",
+                    "weight": 1,
+                    "inheritConfiguration": false,
+                    "configuration": {
+                        "bootstrapServers": "kafka:9092",
+                        "topics" : ["demo"],
+                        "consumer" : {
+                            "enabled": false
+                        },
+                        "producer": {
+                            "enabled": true
+                        }
+                    }
+                }
+            ]
+        }
+    ],
+    "flows": [
+        {
+            "name": "",
+            "selectors": [],
+            "request": [],
+            "response": [],
+            "subscribe": [],
+            "publish": [],
+            "enabled": true
+        }
+    ]
+}
+----
+
+== Quality of Service
+
+When working with asynchronous APIs, quality of service is important. Quality of service defines the guaranteed level of message delivery. For example, a quality of service of "none" means that a given message might be delivered zero, one, or several times. A quality of service of "at-most-once" means that a given message will be delivered zero or one times, with no duplication.
+
+A higher quality of service could lead to lower system performance depending on the endpoint chosen.
+
+The quality-of-service levels are described in the following table.
+
+.Table Quality of service levels
+[cols="1,4"]
+|===
+|Level | Description
+
+| None
+| A given message might be delivered zero, one, or many times. This level allows high throughput and good performance but without guaranteed delivery. After failure or disconnection, the client will only receive messages sent after reconnection.
+
+| Auto (0 or N)
+| A given message might be delivered zero, one, or many times. This level allows a trade-off between performance and delivery guarantee. Delivery is highly dependant on the capabilities supported by the endpoint connector.  After failure or disconnection, after reconnection, the client will resume, if possible, from a previously saved state, although messages duplication could potentially exist.
```

Comment:
```suggestion
| A given message might be delivered zero, one, or many times. This level allows a trade-off between performance and delivery guarantee. Delivery is highly dependent on the capabilities supported by the endpoint connector.  In case of failure or disconnection, after reconnection the client will resume, if possible, from a previously saved state, although duplication of messages could potentially exist.
```